site/docs/view-spec.md

```diff
@@ -0,0 +1,253 @@
+| Required | default-namespace | An array of strings indicating namespace at the time view was created / replaced. Used similar to ‘default-catalog’ above. |
+| Optional | field-aliases | A list of strings of field aliases E.g. a list of alias_name info specified in the following create view statement. `CREATE VIEW v (alias_name COMMENT 'docs', alias_name2, ...) AS SELECT ...` |
+| Optional | field-docs | A list of strings of field comments E.g. a list of ‘comment’ info specified in the following create view statement. `CREATE VIEW v (alias_name COMMENT 'docs', alias_name2, ...) AS SELECT ...` |
+
+
+Note that each version is stored in a separate AVRO file. This is to ensure that the metadata file stays readable in the case the view definition is huge. As a future extension, an engine-agnostic intermediate representation or a serialized abstract syntax tree of the SQL definition may also be stored in each version, exacerbating the problem.
+
+“summary” is a string-string map with the following required keys. Engines may store additional key-value pairs in this map.
+
+| Required/Optional | Key | Value |
+|-------------------|-----|-------|
+| Required | operation |A string value indicating the view operation that caused this metadata to be created. Allowed values are “CREATE” and “REPLACE” |
+| Required | engine-version | A string value indicating the version of the engine that performed the operation (create / replace) |
+
+
+“schema” is an array of “fields” that itself has following fields. Each field represents information about a column projected by the view.
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | id | Id of the field. It is unique across all the fields of the view. |
+| Required | name | Name of the field. |
+| Required |required | Indicates whether the field is a required field. |
+| Required | type | Data type of the field |
+| Optional | doc | Comment describing the field |
+
+“Version-log” is an array of structs describing the log of the versions created. The struct has following fields: 
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | timestamp-ms | Timestamp expressed as ms since epoch for when the version of the view indicated in version-id field below was created |
+| Required | version-id | Version id of the view |
+
+
+Each metadata file is self-sufficient. It contains the history of the last few operations performed on the view and can be used to rollback the view to a previous version (Command to rollback the view is left as future work).
+The table property “version.history.num-entries” controls the number of versions kept. This property defaults to 10.
+
+### Metadata Location Management
+
+The view is created in the default warehouse location with database name and view name in the suffix. An example path is: 
```

Comment:
I think we don't need to mention these defaults, because the default root location is different across different catalog implementations and should not be relied on. If it's managed exactly like table location, we should just say that instead.